numpy/_core/tests/test_regression.py

```diff
@@ -2622,3 +2622,16 @@ def test_vectorize_fixed_width_string(self):
         f = str.casefold
         res = np.vectorize(f, otypes=[arr.dtype])(arr)
         assert res.dtype == "U30"
+
+    def test_repeated_square_consistency(self):
+        # gh-26940
+        buf = np.array([-5.171866611150749e-07 + 2.5618634555957426e-07j,
+                        0, 0, 0, 0, 0])
+        in_vec = buf[:3]
+        # Output vector immediately follows input vector
+        # to reproduce off-by-one in nomemoverlap check.
+        res0 = buf[3:]
+        np.square(in_vec, out=res0)
+        res1 = np.square(in_vec)
+        # Check failed when compiling in clang with "-ffp-contract=off"
```

Comment:
```suggestion
```
Nitpick: I think the comment isn't too helpful since it is ystem specific.  Unfortunatly, it is hard to check that we are not avoiding the branc unnecessarily.

Reply:
Removed